tutorials/03-bayesian-neural-network/index.qmd

```diff
@@ -189,7 +189,7 @@ const nn = StatefulLuxLayer{true}(nn_initial, nothing, st)
     parameters ~ MvNormal(zeros(nparameters), Diagonal(abs2.(sigma .* ones(nparameters))))
 
     # Forward NN to make predictions
-    preds = Lux.apply(nn, xs, vector_to_parameters(parameters, ps))
+    preds = Lux.apply(nn, xs, f32(vector_to_parameters(parameters, ps)))

```

Comment:
@willtebbutt I had to change this line in the Bayesian neural net tutorial to get it to work with Mooncake. Without this, I think it was doing some f32 * f64 matrix multiplication for which `build_rrule` failed: https://github.com/TuringLang/docs/actions/runs/12017000042/job/33498395092#step:8:537 Happy to create an issue at a later point, just wanted to run this past you to make sure it's a sensible fix.

Reply:
Thanks for highlighting this @penelopeysm , and very good to know that this is working well. I think I'm going to have to generalise my implementation of the rule for `LuxLib.Impl.matmul` to handle mixed precision inputs, which is not something I thought would be necessary!